src/wasm.cc

```diff
@@ -472,6 +472,11 @@ bool WasmHandleBase::canary(const std::shared_ptr<PluginBase> &plugin,
   if (this->wasm() == nullptr) {
     return false;
   }
+  std::string plugin_key(plugin->key());
```

Comment:
nit: can `plugin->key()` be used directly throughout this function? Would avoid a needless copy, and it shouldn't change.

Reply:
Done. It turns out key() is pre-computed already so no need to store it again.